vega_sim/service.py

```diff
@@ -1292,30 +1292,7 @@ def order_status(
             Optional[vega.Order], the requested Order object or None if nothing found
         """
         return data_raw.order_status(
-            order_id=order_id, data_client=self.trading_data_client, version=version
-        )
-
-    def order_status_by_reference(
```

Comment:
Can we not keep this but just use the new query pattern?

Reply:
I removed this one as it wasn't used anywhere in the codebase and the v1 method `OrderByReference` doesn't have a direct equivalent in v2.

The closest v2 method is `ListOrders` which already has a method in `service.py` called `list_orders`.

 I have just realised the `service.list_orders` methods is missing an optional `reference` arg to pass through to `data.list_orders` method though. If you are happy to leave out `order_status_by_reference` and just use `list_orders` I'll add that arg in?

Reply:
Yeah that sounds good to me 👍